src/persist-client/src/fetch.rs

```diff
@@ -356,6 +356,23 @@ where
     Ok(part)
 }
 
+/// This represents the lease of a seqno. It's generally paired with some external state,
+/// like a hollow part: holding this lease indicates that we may still want to fetch that part,
+/// and should hold back GC to keep it around.
+///
+/// Generally the state and lease are bundled together, as in [LeasedBatchPart]... but sometimes
+/// it's necessary to handle them separately, so this struct is exposed as well. Handle with care.
+#[derive(Clone, Debug, Default)]
+pub struct Lease(Arc<()>);
+
+impl Lease {
+    /// Returns the number of live copies of this lease... aside from this one.
+    pub fn copies(&self) -> usize {
+        // NB: the strong count is always > 0 for a live arc.
+        Arc::strong_count(&self.0) - 1
```

Comment:
This is only used in one place to compare to 0. Personally, I don't find the complexity of the NB and such to be worth it vs just returning the strong_count and comparing to 1 instead of 0.

An alternative is to store a `weak` ref in the ReadHandle's map, and then strong_count is exactly equal to the number of outstanding leases. If it ends up not being a terribly fiddly diff (e.g. gotta swap it out if the weak upgrade fails), I think I prefer this to both what you have and "compare to 1"

Ultimately, I think any of these three are workable though, so will leave the call to you

Reply:
I did try the weak version initially... and while it seemed to work, the code was much less obvious.